### PR TITLE
Add Stronghold migration instructions

### PR DESCRIPTION
**Pull request by eike-hass**
# Description of change

Add migration instructions for Stronghold v2 to Identity 0.6.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Documentation Enhancement

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [X] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [X] I have performed a self-review of my own changes
- [ ] I have made sure that added/changed links still work
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by PhilippGackstatter — APPROVED** [review]
We could do what Abdul suggested and link to the function that can be used for the migration to make the CLI tool seem less mandatory to be used. But overall it's fine.

File: iota/docs/identity.rs/v0.5.0/src/partials/_announcement.mdx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+:::info
+The 0.5 version of IOTA Identity is deprecated. Please upgrade to [Identity 0.6 for Chrysalis Networks](TBD) and follow the migration instructions for upgrading Stronghold Snapshot files [here](TBD), or upgrade to [Identity 1.0 for Stardust Networks](/identity.rs/tutorials/migrate_stronghold/).
+:::
```
Comment (Dr-Electron) [inline iota/docs/identity.rs/v0.5.0/src/partials/_announcement.mdx:3]:
```suggestion
:::info

The 0.5 version of IOTA Identity is deprecated. Please upgrade to [Identity 0.6 for Chrysalis Networks](TBD) and follow the migration instructions for upgrading Stronghold Snapshot files [here](TBD), or upgrade to [Identity 1.0 for Stardust Networks](/identity.rs/tutorials/migrate_stronghold/).

:::
```

File: iota/docs/identity.rs/v0.6.0/docs/tutorials/migrate_stronghold.mdx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+import Announcement from '../../src/partials/_announcement.mdx';
+
+<Announcement />
+
+With the Rust 0.6.3 and Wasm 0.6.1 version of IOTA Identity, Stronghold was updated to [v2](https://github.com/iotaledger/stronghold.rs/releases/tag/iota-stronghold-v2.0.0). The new Stronghold release, besides many security enhancements, introduces an updated Snapshot format and requires old Snapshots to be migrated to the new format.
+
+:::info
+Stronghold Snapshots v1 must be migrated to v2 before they can be used with Identity Rust 0.6.3 and Wasm 0.6.1.
+:::
```
Comment (Dr-Electron) [inline iota/docs/identity.rs/v0.6.0/docs/tutorials/migrate_stronghold.mdx:18]:
```suggestion
:::info

Stronghold Snapshots v1 must be migrated to v2 before they can be used with Identity Rust 0.6.3 and Wasm 0.6.1.

:::
```

File: iota/docs/identity.rs/v0.6.0/docs/tutorials/migrate_stronghold.mdx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+To migrate Stronghold Snapshot files from the v1 to the v2 format the following instructions using IOTA SDK for Rust or Nodejs can be utilized.
+
+:::danger
+After verifying that your new Stronghold Snapshot works as intended, make sure to securely delete the old Stronghold Snapshot and **all** of your backups, and replace them with backups of the migrated Stronghold Snapshot, if desired.
+:::
```
Comment (Dr-Electron) [inline iota/docs/identity.rs/v0.6.0/docs/tutorials/migrate_stronghold.mdx:24]:
```suggestion
:::danger

After verifying that your new Stronghold Snapshot works as intended, make sure to securely delete the old Stronghold Snapshot and **all** of your backups, and replace them with backups of the migrated Stronghold Snapshot, if desired.

:::
```

File: iota/docs/identity.rs/v0.6.0/src/partials/_announcement.mdx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+:::info
+The latest upgrade of Stronghold in Identity Rust 0.6.3 and Wasm 0.6.1 requires a migration of Snapshot files. You can find migration instructions [here](/identity.rs/tutorials/migrate_stronghold/).
+:::
```
Comment (Dr-Electron) [inline iota/docs/identity.rs/v0.6.0/src/partials/_announcement.mdx:3]:
```suggestion
:::info

The latest upgrade of Stronghold in Identity Rust 0.6.3 and Wasm 0.6.1 requires a migration of Snapshot files. You can find migration instructions [here](/identity.rs/tutorials/migrate_stronghold/).

:::
```

**Review by Dr-Electron — COMMENTED** [review]
Just some formatting stuff. Docusaurus recommends to do it this way to not create any potential issues

**Comment by eike-hass** [comment]
> We could do what Abdul suggested and link to the function that can be used for the migration to make the CLI tool seem less mandatory to be used. But overall it's fine.

Both would be best probably.

File: iota/docs/identity.rs/v0.6.0/docs/tutorials/migrate-stronghold.mdx [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+:::
+
+## Rust
+
+To migrate a v2 Stronghold Snapshot from Identity 0.6.2 or earlier in Rust to a v3 Stronghold Snapshot that can be used with Identity 0.6.3 Rust or later, follow these steps:
+
+- Create a new crate with `cargo new --bin stronghold-migration && cd stronghold-migration`.
+- Run `cargo add iota-sdk --features stronghold`.
+- Run `cargo add anyhow`.
+- Copy the code from [rust migration tool](#Rust-Migration-Tool) into `main.rs`.
```
Comment (abdulmth) [inline iota/docs/identity.rs/v0.6.0/docs/tutorials/migrate-stronghold.mdx:37]:
```suggestion
- Copy the code from [rust migration tool](#rust-migration-tool) into `main.rs`.
```

**Review by Dr-Electron — APPROVED** [review]
I was forced to  do this. Help.

**Pull request merged**